cpp/include/legate_dataframe/stream_compaction.hpp

```diff
@@ -44,4 +58,21 @@ class ApplyBooleanMaskTask : public Task<ApplyBooleanMaskTask, OpCode::ApplyBool
  */
 LogicalTable apply_boolean_mask(const LogicalTable& tbl, const LogicalColumn& boolean_mask);
 
+/**
+ * @brief Find distinct (unique) rows in a logical table.
+ *
+ * For rows not listed as keys any value is possible, order is not guaranteed.
+ *
+ * @param tbl The table to distinct.
+ * @param keys The column names to distinct by.
+ * @param high_cardinality Whether the table is assumed to have a high cardinality.
+ * If false (default), the cardinality is assumed to be low in which case we can save
+ * communication by doing a local distinct before shuffling data.  If the cardinality is high,
```

Comment:
I think the default strategy is much more robust and you might not need the paramenter. Have you tested any performance improvement for high cardinality inputs?